### PR TITLE
Post edit/delete

### PR DESCRIPTION
**Pull request by neetukumari4858**
Edit & Delete Post
- Edit Post
I can click on the "Edit Post" button which will open the same component with the textbox where we can edit a post & save it.
- Delete Post
I can click on the "Delete Post" button which will delete a post.

### CODE REVIEW

**Review by MonikaJhingan — APPROVED** [review]
Looks good to be merged

File: src/redux/asyncThunks/postThunk.js [hunk 1]
```diff
@@ -47,4 +47,64 @@ const dislikePost = createAsyncThunk(
...
+    try {
+      const response = await axios.post(
+        '/api/posts',
+        { postData },
+        {
+          headers: { authorization: token },
+        },
+      )
+      const data = { data: response.data }
+      // console.log(response)
```
Comment (tusharkandpal) [inline src/redux/asyncThunks/postThunk.js]: please remove this comment!

File: src/redux/asyncThunks/postThunk.js [hunk 2]
```diff
@@ -47,4 +47,64 @@ const dislikePost = createAsyncThunk(
...
+const editPost = createAsyncThunk(
+  'posts/editPost',
+  async ({ postDetail, postData, token }, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `/api/posts/edit/${postDetail._id}`,
+        { postData },
+        { headers: { authorization: token } },
+      )
+      console.log(response,"edit")
```
Comment (tusharkandpal) [inline src/redux/asyncThunks/postThunk.js:97]: please remove this console.!

File: src/redux/asyncThunks/postThunk.js [hunk 3]
```diff
@@ -47,4 +47,64 @@ const dislikePost = createAsyncThunk(
...
+      const data = { data: response.data, status: response.status }
+      return data
+    } catch (error) {
+      console.error(error)
```
Comment (tusharkandpal) [inline src/redux/asyncThunks/postThunk.js:101]: this too

File: src/redux/slices/postSlice.js [hunk 4]
```diff
@@ -33,7 +33,25 @@ export const postSlice = createSlice({
     [dislikePost.rejected]: (action) => {
       console.log(action)
```
Comment (tusharkandpal) [inline src/redux/slices/postSlice.js:34]: here too

File: src/redux/slices/postSlice.js [hunk 5]
```diff
@@ -33,7 +33,25 @@ export const postSlice = createSlice({
...
+      state.posts = action.payload.data.posts
+    },
+    [deletePost.rejected]: (action) => {
+      console.log(action)
+    },
+    [editPost.fulfilled]: (state, action) => {
+      state.posts = action.payload.data.posts
+    },
+    [editPost.rejected]: (action) => {
+      console.log(action)
```
Comment (tusharkandpal) [inline src/redux/slices/postSlice.js:53]: +1

File: src/redux/slices/postSlice.js [hunk 6]
```diff
@@ -33,7 +33,25 @@ export const postSlice = createSlice({
...
+      state.posts = action.payload.data.posts
+    },
+    [createPost.rejected]: (action) => {
+      console.log(action)
+    },
+    [deletePost.fulfilled]: (state, action) => {
+      state.posts = action.payload.data.posts
+    },
+    [deletePost.rejected]: (action) => {
+      console.log(action)
```
Comment (tusharkandpal) [inline src/redux/slices/postSlice.js:47]: +1

File: src/redux/slices/postSlice.js [hunk 7]
```diff
@@ -33,7 +33,25 @@ export const postSlice = createSlice({
     [dislikePost.rejected]: (action) => {
       console.log(action)
     },
-  },
+    },
+    [createPost.fulfilled]: (state, action) => {
+      state.posts = action.payload.data.posts
+    },
+    [createPost.rejected]: (action) => {
+      console.log(action)
```
Comment (tusharkandpal) [inline src/redux/slices/postSlice.js:41]: +1 I think you'll handle it afterwards.

**Review by tusharkandpal — APPROVED** [review]
looks good  👍🏻  ...just format a little bit &  please remove console & comments before raising PR .

File: src/redux/asyncThunks/postThunk.js (hunk 1 above)
Comment (neetukumari4858) [reply src/redux/asyncThunks/postThunk.js]: okey I will remove

File: src/redux/asyncThunks/postThunk.js (hunk 3 above)
Comment (neetukumari4858) [reply src/redux/asyncThunks/postThunk.js:101]: here I need console to check the error message that's why I used

**Pull request merged**